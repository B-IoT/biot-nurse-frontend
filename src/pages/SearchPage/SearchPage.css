.search-page {
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  padding-top: 50px;
  padding-bottom: 50px;
}

.search-title {
  text-align: center;
}

.result-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 80%;
  margin-top: 20px;
}

.item-container {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  border-radius: 30px;
  height: 160px;
  width: 160px;
  margin: 20px;
  transform: scale(0.9);
  -webkit-transform-origin: center;
}

.text-container {
  display: flex;
  position: relative;
  flex-direction: column;
  will-change: transform;
}

.item-icon {
  position: relative;
  height: 60px;
}

.item-text {
  position: relative;
  text-align: center;
  margin-top: 12px;
  padding-left: 10px;
  padding-right: 10px;
}
