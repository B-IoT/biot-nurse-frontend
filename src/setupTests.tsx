// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter } from 'react-router-dom';
import React, { ComponentType } from 'react';

const queryClient = new QueryClient();
export const router = (children: any) => {
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>{children}</BrowserRouter>
    </QueryClientProvider>
  );
};

const Wrapper: React.ComponentType = ({ children }) => {
  return router(children);
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper as ComponentType, ...options });

global.URL.createObjectURL = jest.fn();

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
