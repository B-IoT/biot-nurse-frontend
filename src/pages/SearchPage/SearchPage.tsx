import { useEffect, useState } from 'react';
import Input from '../../components/input/Input/Input';

import './SearchPage.css';
import { Category, getIconPath, simplifyText } from '../../utils/items';
import { useQuery } from 'react-query';
import { getCategories } from '../../api/api';
import LogOut from '../../components/button/LogOut/LogOut';
import { MAP_PATH } from '../../App';
import Button from '../../components/button/Button/Button';
import { Scale } from '../../utils/animations';
import { Link } from 'react-router-dom';

/**
 * The search page where the user can browse the different item categories.
 */
export default function SearchPage() {
  const [categories, setCategories] = useState([] as Category[]);
  const { data } = useQuery('categories', getCategories);

  useEffect(() => {
    if (data) {
      setCategories(data);
    }
  }, [data]);

  const [keyword, setKeyword] = useState('');
  const [buttons, setButtons] = useState([<div />]);
  useEffect(
    () =>
      setButtons(
        categories
          .filter(
            (category) =>
              keyword === '' ||
              simplifyText(category.name).includes(simplifyText(keyword))
          )
          .map((category) => (
            <Link
              key={category.id}
              className="item-container"
              to={{ pathname: MAP_PATH, state: { category } }}
              style={{ textDecoration: 'none' }}
            >
              <Button
                onClick={() => null}
                width={210}
                height={210}
                borderRadius={50}
                style={{}}
                data-testid="item-button"
              >
                <Scale className="text-container clear">
                  <img
                    className="item-icon"
                    src={getIconPath(category.name)}
                    alt="Item icon"
                  />
                  <div className="item-text axiforma-medium-blue-18px">
                    {' '}
                    {category.name}{' '}
                  </div>
                </Scale>
              </Button>
            </Link>
          ))
      ),
    [categories, keyword]
  );

  return (
    <div className="search-page">
      <h1 className="search-title axiforma-bold-blue-70px">
        {'Que cherchez-vous ?'}
      </h1>
      <Input
        setKeyword={setKeyword}
        defaultText="Rechercher"
        width={550}
        style={{ marginTop: 50 }}
        isPassword={false}
        enterHandler={() => null}
      />
      <div className="result-grid">{buttons}</div>
      <LogOut />
    </div>
  );
}
