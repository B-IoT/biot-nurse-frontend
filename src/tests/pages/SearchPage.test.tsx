import { render, router } from '../../setupTests';
import SearchPage from '../../pages/SearchPage/SearchPage';
import renderer from 'react-test-renderer';

it('Search page renders without crashing', () => {
  render(<SearchPage />);
});

it('Search page matches snapshot', () => {
  const rendered = renderer.create(router(<SearchPage />));
  expect(rendered).toMatchSnapshot();
});
