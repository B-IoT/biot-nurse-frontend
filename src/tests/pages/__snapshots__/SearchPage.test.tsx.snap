// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page matches snapshot 1`] = `
<div
  className="search-page"
>
  <h1
    className="search-title font-axiforma-bold text-blue text-title"
  >
    Que cherchez-vous ?
  </h1>
  <div
    className="search-bar"
    style={
      Object {
        "marginTop": 25,
        "width": 300,
      }
    }
  >
    <div
      className="search-blur"
      style={
        Object {
          "width": 310,
        }
      }
    >
      <div
        className="search-mask"
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="search-shadow"
          style={
            Object {
              "width": 350,
            }
          }
        />
        <div
          className="search-light"
          style={
            Object {
              "width": 350,
            }
          }
        />
      </div>
    </div>
    <input
      className="search-text font-axiforma-light text-blue text-small"
      data-testid="input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Rechercher"
      style={
        Object {
          "width": 250,
        }
      }
      type=""
    />
  </div>
  <div
    className="result-grid"
  >
    <div />
  </div>
  <div
    className="log-out"
  >
    <div
      className="button css-1i20ack-Hover e128ygwg0"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "height": 60,
          "marginBottom": 30,
          "width": 60,
        }
      }
    >
      <div
        className="button-pressed css-1gvt6rq-FadeIn e128ygwg3"
        style={
          Object {
            "borderRadius": 60,
            "filter": "blur(4px)",
            "height": "calc(60px + 0.5*5px)",
            "width": "calc(60px + 0.5*5px)",
          }
        }
      />
      <div
        className="button-unpressed css-1ypgtc4-FadeOut e128ygwg2"
      >
        <div
          className="neumorphic-container"
          style={
            Object {
              "height": 60,
              "width": 60,
            }
          }
        >
          <div
            className="neumorphic-div"
            style={
              Object {
                "background": "var(--darker-shadow)",
                "borderRadius": 60,
                "filter": "blur(2px)",
                "height": "calc(100% + 5px)",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "calc(100% + 5px)",
              }
            }
          />
          <div
            className="neumorphic-div"
            style={
              Object {
                "background": "var(--lighter-light)",
                "borderRadius": 60,
                "bottom": 0,
                "filter": "blur(2px)",
                "height": "calc(100% + 5px)",
                "position": "absolute",
                "right": 0,
                "width": "calc(100% + 5px)",
              }
            }
          />
          <div
            className="neumorphic-div neumorphic-ambiant"
            style={
              Object {
                "background": "linear-gradient(157.37deg, var(--light) 13.29%, var(--shadow) 84.76%)",
                "borderRadius": 60,
                "filter": "blur(2px)",
                "height": "100%",
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <div
            className="icon-container clear css-zcw12y-Scale e128ygwg1"
          >
            <img
              alt="Button icon"
              className="round-icon"
              src="navbarIcons/logOut.svg"
            />
          </div>
        </div>
      </div>
      <div
        className="icon-container clear css-zcw12y-Scale e128ygwg1"
      >
        <img
          alt="Button icon"
          className="round-icon"
          src="navbarIcons/logOut.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
